File: visa-dev-samplecode-master/vdp-node/test/pymntacntattrinqry/fundstransferattributetest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Payment Account Attributes Funds Transfer Test', function() {
    var visaAPIClient = new VisaAPIClient();
    var fundsTransferInquiry = JSON.stringify({
          "acquirerCountryCode": "840",
          "acquiringBin": "408999",
          "primaryAccountNumber": "4957030420210512",
          "retrievalReferenceNumber": "330000550000",
          "systemsTraceAuditNumber": "451006"
        });
    
    it('Funds Transfer Inquiry Test',function(done) {
        this.timeout(10000);
        var baseUri = 'paai/';
        var resourcePath = 'fundstransferattinq/v1/cardattributes/fundstransferinquiry';
        visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, fundsTransferInquiry, 'POST', {}, 
        function(err, responseCode) {
            if(!err) {
                assert.equal(responseCode, 200);
            } else {
                assert(false);
            }
            done();
        });        
    });
});

File: vdp-ruby/test/pymntacntattrinqry/funds_transfer_attribute_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class FundsTransferAttributesTest < Test::Unit::TestCase
  def setup
    @visa_api_client = VisaAPIClient.new
    @fundsTransferInquiry ='''{
    "acquirerCountryCode": "840",
    "acquiringBin": "408999",
    "primaryAccountNumber": "4957030420210512",
    "retrievalReferenceNumber": "330000550000",
    "systemsTraceAuditNumber": "451006"
  }'''
  end
  
  def test_fundsTransferInquiry
    base_uri = 'paai/'
    resource_path = 'fundstransferattinq/v1/cardattributes/fundstransferinquiry'
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Funds Transfer Inquiry test", "post", @fundsTransferInquiry)
    assert_equal("200", response_code, "Funds Transfer Inquiry test failed")
  end
end
